src/components/Game/Game.jsx

```diff
@@ -105,19 +107,36 @@ const Game = ({ gameId, playerId }) => {
     });
     // catch illegal move
     if (move === null) return false;
-    // --> game ending scenarios will need to be handled.
-    if (game.game_over()) setGameData(prev => ({ ...prev, complete: true }));
-    if (game.in_draw()) setGameData(prev => ({ ...prev, draw: true }));
-    // <--
-    socket.emit('make_move', { current_fen: game.fen(), game_id: gameId });
+    const gameComplete = checkGameCompletion(game);
+    if (!gameComplete) socket.makeMove(game.fen(), gameId);
     return true;
   }
 
+  function checkGameCompletion(currentGame) {
+    if (currentGame.game_over()) {
+      if (currentGame.in_draw()) {
+        setGameData(prev => ({ ...prev, outcome: 'draw' }));
+      } else if (currentGame.in_stalemate()) {
+        setGameData(prev => ({ ...prev, outcome: 'stalemate' }));
+      } else if (currentGame.in_checkmate()) {
+        setGameData(prev => ({
+          ...prev,
+          outcome: 'checkmate',
+          champion:
+            game.turnColor === 'white'
+              ? gameData.whitePlayerId
+              : gameData.blackPlayerId,
+        }));
+      }
+      return true;
+    }
+  }
+
   return error ? (
     <LoadError />
   ) : game ? (
     <Chessboard
-      position={game.fen()}Í
```

Comment:
We will need to handle a merge conflict since this doesn't include the game info component.